### PR TITLE
SSVNetwork audit recommendations

### PR DESCRIPTION
**Pull request by mtabasco**
- [x] LIBRARIES
Set the visibility of RegisterAuth.SSV_STORAGE_POSITION to private .
Set the visibility of SSVStorage.SSV_STORAGE_POSITION to private .
Set the visibility of SSVStorageProtocol.SSV_STORAGE_POSITION to private
- [x] To facilitate logging it is recommended to index address parameters within events. Therefore the (other) address parameters in indexed keyword should be added to the CoreLib.ModuleUpgraded()
- [x] Add `getVersion` function to SSVNetwork contract

**ABI changes**
function getVersion() external pure returns (string memory version)

**Upgrade plan**

After updating SSVClusters, SSVOperators and SSVDAO, upgrade SSVNetwork.

### CODE REVIEW

File: contracts/libraries/CoreLib.sol [hunk 1]
```diff
@@ -4,7 +4,7 @@ pragma solidity 0.8.18;
 import "./SSVStorage.sol";
 
 library CoreLib {
-    event ModuleUpgraded(SSVModules moduleId, address moduleAddress);
+    event ModuleUpgraded(SSVModules indexed moduleId, address moduleAddress);
```
Comment (andrew-blox) [inline contracts/libraries/CoreLib.sol:7]: This wont make any effect on the devs right?

File: contracts/libraries/RegisterAuth.sol [hunk 2]
```diff
@@ -1,17 +1,13 @@
...
-import "@openzeppelin/contracts/utils/Counters.sol";
-
 struct Authorization {
     bool registerOperator;
     bool registerValidator;
 }
 
 library RegisterAuth {
-    uint256 constant SSV_STORAGE_POSITION = uint256(keccak256("ssv.network.storage.auth")) - 1;
+    uint256 constant private SSV_STORAGE_POSITION = uint256(keccak256("ssv.network.storage.auth")) - 1;
```
Comment (andrew-blox) [inline contracts/libraries/RegisterAuth.sol:10]: Its upgradeable right?

File: contracts/libraries/SSVStorage.sol [hunk 3]
```diff
@@ -37,7 +36,7 @@ struct StorageData {
 }
 
 library SSVStorage {
-    uint256 constant SSV_STORAGE_POSITION = uint256(keccak256("ssv.network.storage.main")) - 1;
+    uint256 constant private SSV_STORAGE_POSITION = uint256(keccak256("ssv.network.storage.main")) - 1;
```
Comment (andrew-blox) [inline contracts/libraries/SSVStorage.sol:39]: Its upgradeable right?

File: contracts/libraries/SSVStorageProtocol.sol [hunk 4]
```diff
@@ -33,7 +31,7 @@ struct StorageProtocol {
 }
 
 library SSVStorageProtocol {
-    uint256 constant SSV_STORAGE_POSITION = uint256(keccak256("ssv.network.storage.protocol")) - 1;
+    uint256 constant private SSV_STORAGE_POSITION = uint256(keccak256("ssv.network.storage.protocol")) - 1;
```
Comment (andrew-blox) [inline contracts/libraries/SSVStorageProtocol.sol:34]: Its upgradeable right?

**Review by andrew-blox — APPROVED** [review]

File: contracts/libraries/CoreLib.sol (hunk 1 above)
Comment (mtabasco) [reply contracts/libraries/CoreLib.sol:7]: No, they can use it as usual. We can now filter by `moduleId` to search more easily when a module was upgraded. No need to add `msg.sender` as its implicit in the transaction.

File: contracts/libraries/SSVStorage.sol (hunk 3 above)
Comment (mtabasco) [reply contracts/libraries/SSVStorage.sol:39]: yes

File: contracts/libraries/SSVStorageProtocol.sol (hunk 4 above)
Comment (mtabasco) [reply contracts/libraries/SSVStorageProtocol.sol:34]: yes

**Review by lior-blox — APPROVED** [review]

**Pull request merged**